src/Controller/Component/UsersAuthComponent.php

```diff
@@ -114,6 +118,38 @@ protected function _initAuth()
     }
 
     /**
+     * Check Public Access
+     *
+     * ### Usage
+     *
+     * Your permssions config should return '*' or 'public' for role(s) allowed.
+     *
+     * Ex. In permissions.php for SimpleRbacAuthorize
+     * ```php
+     * 'Users.SimpleRbac.permissions' => [
+     *     [
+     *         'role' => '*',
+     *         'controller' => ['Pages'],
+     *         'action' => ['other', 'display'],
+     *         'allowed' => true,
+     *     ]];
+     *
+     * @return bool
+     */
+    public function isPublicAuthorized()
+    {
+        $isAuthorized = null;
+        $controller = $this->_registry->getController();
+        if (empty($controller->Auth->user())) {
+            $isAuthorized = $controller->Auth->isAuthorized(['role' => 'public'], $controller->request);
+            if ($isAuthorized === true) {
+                $controller->Auth->allow();
```

Comment:
That is what this is doing in effect, because this function is called on a per-request basis.  So allow() only applies to the current action being requested.